src/platforms/evdev-rs/src/lib.rs

```diff
@@ -1,25 +1,1404 @@
+        input_sink: *mut ffi::InputSink,
+        event_builder: *mut ffi::EventBuilder,
+    ) {
+        self.tx
+            .send(ThreadCommand::Start(
+                self.device_id,
+                InputSinkPtr(std::ptr::NonNull::new(input_sink).unwrap()),
+                EventBuilderPtr(std::ptr::NonNull::new(event_builder).unwrap()),
+            ))
+            .unwrap();
+        nix::unistd::write(&self.wfd, &[0u8]).unwrap();
+    }
+
+    pub fn stop(&mut self) {}
+
+    pub fn get_device_info(&self) -> Box<InputDeviceInfoRs> {
+        let (tx, rx) = create_device_info_rs_channel();
+        self.tx
+            .send(ThreadCommand::GetDeviceInfo(self.device_id, tx))
+            .unwrap();
+        nix::unistd::write(&self.wfd, &[0u8]).unwrap();
+        let device_info = rx.recv().unwrap();
+        Box::new(device_info)
+    }
+
+    pub fn get_pointer_settings(&self) -> Box<ffi::PointerSettingsRs> {
+        let (tx, rx) = create_pointer_settings_rs_channel();
+        self.tx
+            .send(ThreadCommand::GetPointerSettings(self.device_id, tx))
+            .unwrap();
+        nix::unistd::write(&self.wfd, &[0u8]).unwrap();
+        let pointer_settings = rx.recv().unwrap();
+        Box::new(pointer_settings)
+    }
+
+    pub fn set_pointer_settings(&self, settings: &ffi::PointerSettingsC) {
+        self.tx
+            .send(ThreadCommand::SetPointerSettings(
+                self.device_id,
+                settings.clone(),
+            ))
+            .unwrap();
+        nix::unistd::write(&self.wfd, &[0u8]).unwrap();
+    }
+}
+
+impl ffi::PointerSettingsRs {
+    pub fn empty() -> Self {
+        ffi::PointerSettingsRs {
+            is_set: false,
+            handedness: 0,
+            cursor_acceleration_bias: 0.0,
+            acceleration: ffi::MirPointerAcceleration::mir_pointer_acceleration_none.repr,
+            horizontal_scroll_scale: 1.0,
+            vertical_scroll_scale: 1.0,
+        }
+    }
+}
+
+#[cxx::bridge(namespace = "mir::input::evdev_rs")]
+mod ffi {
+    // Note(mattkae): The device capability enum MUST be in its expanded
+    // form instead of using a bit shift (e.g. 1 << 3) because cxx
+    // currently does not support bit shifts in enums. This will result
+    // in the error: "enums with non-integer literal discriminants are
+    // not supported yet".
+    #[repr(u32)]
+    #[derive(Debug, Copy, Clone, PartialEq, Eq)]
+    enum DeviceCapability {
+        unknown = 0,
+        pointer = 2,
+        keyboard = 4,
+        touchpad = 8,
+        touchscreen = 16,
+        gamepad = 32,
+        joystick = 64,
+        switch_ = 128,
+        multitouch = 256,
+        alpha_numeric = 512,
+    }
+
+    #[repr(i32)]
+    #[derive(Debug, Copy, Clone, PartialEq, Eq)]
+    enum MirKeyboardAction {
+        mir_keyboard_action_up,
+        mir_keyboard_action_down,
+        mir_keyboard_action_repeat,
+        mir_keyboard_action_modifiers,
+        mir_keyboard_actions,
+    }
```

Comment:
No way to bring these from `.h`?

How do we ensure these are equivalent to those…?